types/ndarray/index.d.ts

```diff
@@ -34,7 +35,14 @@ declare namespace ndarray {
         T: NdArray<D>;
     }
 
-    type Data<T = any> = T[] | TypedArray;
+    type Data<T = any> = T extends number
+        ? T[] | TypedArray
+        : T extends bigint
+        ? T[] | BigInt64Array | BigUint64Array
+        : T extends bigint | number
+        ? T[] | TypedArray | BigInt64Array | BigUint64Array
```

Comment:
Nice, it works really well. I don't even think you need this `T extends bigint | number` conditional branch.

Reply:
You are correct! Fixed -- thanks!